pkg/render/node.go

```diff
@@ -1820,9 +1820,9 @@ func (c *nodeComponent) hostPathInitContainer() corev1.Container {
 		ImagePullPolicy: ImagePullPolicy(),
 		Command:         []string{"sh", "-c", "calico-node -hostpath-init"},
 		Env: []corev1.EnvVar{
-			{Name: "NODE_USER_ID", Value: "999"},
+			{Name: "NODE_USER_ID", Value: fmt.Sprintf("%d", securitycontext.GetNonRootUID())},
 		},
-		SecurityContext: securitycontext.NewRootContext(false),
+		SecurityContext: securitycontext.NewRootContext(true),
```

Comment:
Why is this changed to `true` (privileged)?

Reply:
I see issues when the init container does `mkdirAll /var/log/calico/cni`.

Reply:
@sridhartigera do you know why this was working before then? Or was it never working? Or is this only needed for BPF mode? Or something about your setup?